main.c

```diff
@@ -1256,10 +1256,11 @@ int main(int argc, char *argv[], char *envp[])
       struct EventMailbox em = { m };
       notify_send(dlg->notify, NT_MAILBOX, NT_MAILBOX_SWITCH, &em);
 
-      mutt_index_menu(dlg, m);
+      m = mutt_index_menu(dlg, m);
```

Comment:
Could this be assigned to a separate variable?

Reply:
Actually, no.

The original Mailbox, `m`, might get closed by the Index, e.g. if the user changes Account.
Or it might not.

When `mutt_index_menu()` finishes, it returns the 'current' Mailbox, which we must `mailbox_free()`.